database/mysql/mysql.go

```diff
@@ -0,0 +1,58 @@
+package mysql
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	"gorm.io/driver/mysql"
+	"gorm.io/gorm"
+	"gorm.io/gorm/logger"
+	"gorm.io/plugin/dbresolver"
+)
+
+// NewMySQLDB creates the mysql master/slaves cluster.
+func NewMySQLDB(cfg Config) (*gorm.DB, error) {
+	dsnTemplate := "%s:%s@tcp(%s:%d)/%s?charset=utf8mb4&parseTime=True&loc=Local"
+	masterDSN := fmt.Sprintf(dsnTemplate,
+		cfg.Master.Username,
+		cfg.Master.Password,
+		cfg.Master.Host,
+		cfg.Master.Port,
+		cfg.Master.DBName,
+	)
+	db, err := gorm.Open(mysql.Open(masterDSN), &gorm.Config{
+		Logger: logger.Default.LogMode(logger.LogLevel(cfg.LogLevel)),
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "open master mysql")
+	}
+
+	var slaveDSNs []gorm.Dialector
+	for _, slave := range cfg.Slaves {
+		slaveDSN := fmt.Sprintf(dsnTemplate,
+			slave.Username,
+			slave.Password,
+			slave.Host,
+			slave.Port,
+			slave.DBName,
+		)
+		slaveDSNs = append(slaveDSNs, mysql.Open(slaveDSN))
+	}
+
+	dbResolverCfg := dbresolver.Config{
+		Sources:  []gorm.Dialector{mysql.Open(masterDSN)},
```

Comment:
do we need to open master again here? Use db?

Reply:
db doesn't implement GORM database dialector interface.